src/index/scanners/default.rs

```diff
@@ -504,3 +362,80 @@ impl SearchBuilder for DefaultBuilder {
     }
 }
+
+fn seq_filter<T, F>(heap: BinaryHeap<T>, f: F) -> BinaryHeap<T>
+where
+    F: Fn(&T) -> bool,
+    T: Ord,
+{
+    match enable_prefilter() {
+        true => heap.into_iter().filter(f).collect(),
+        false => heap,
+    }
+}
+
+type SeqElement<'a> = (
+    (Reverse<Distance>, AlwaysEqual<()>),
+    AlwaysEqual<&'a mut (NonZero<u64>, u16, &'a mut [u32])>,
+);
+
+#[allow(clippy::too_many_arguments)]
+fn rerank_wrapper<'a, O: Operator<Vector = VectOwned<T>>, R, T>(
+    vector: O::Vector,
+    index: &'a R,
+    opfamily: Opfamily,
+    fetch: impl FnMut(NonZero<u64>) -> Option<VectOwned<T>> + 'a,
+    results: BinaryHeap<SeqElement<'a>>,
+    method: RerankMethod,
+    io_rerank: Io,
+) -> Box<dyn Iterator<Item = (f32, NonZero<u64>)> + 'a>
+where
+    R: RelationRead + RelationPrefetch + RelationReadStream,
+{
+    match (method, io_rerank) {
+        (RerankMethod::Index, Io::Plain) => {
+            let prefetcher = PlainPrefetcher::<_, BinaryHeap<_>>::new(index, results);
+            Box::new(
+                rerank_index::<O, _, _>(vector, prefetcher)
+                    .map(move |(distance, payload)| (opfamily.output(distance), payload)),
+            )
+        }
+        (RerankMethod::Index, Io::Simple) => {
+            let prefetcher = SimplePrefetcher::<'a, R, BinaryHeap<_>>::new(index, results);
+            Box::new(
+                rerank_index::<O, _, _>(vector, prefetcher)
+                    .map(move |(distance, payload)| (opfamily.output(distance), payload)),
+            )
+        }
+        (RerankMethod::Index, Io::Stream) => {
+            let prefetcher =
+                StreamPrefetcher::<_, BinaryHeap<_>>::new(index, results, Hints::default());
+            Box::new(
+                rerank_index::<O, _, _>(vector, prefetcher)
+                    .map(move |(distance, payload)| (opfamily.output(distance), payload)),
+            )
+        }
+        (RerankMethod::Heap, Io::Plain) => {
+            let prefetcher = PlainPrefetcher::<_, BinaryHeap<_>>::new(index, results);
+            Box::new(
+                rerank_heap::<O, _, _>(vector, prefetcher, fetch)
+                    .map(move |(distance, payload)| (opfamily.output(distance), payload)),
+            )
+        }
+        (RerankMethod::Heap, Io::Simple) => {
+            let prefetcher = PlainPrefetcher::<_, BinaryHeap<_>>::new(index, results);
+            Box::new(
+                rerank_heap::<O, _, _>(vector, prefetcher, fetch)
+                    .map(move |(distance, payload)| (opfamily.output(distance), payload)),
+            )
+        }
+        (RerankMethod::Heap, Io::Stream) => {
+            let prefetcher =
+                StreamPrefetcher::<_, BinaryHeap<_>>::new(index, results, Hints::default());
+            Box::new(
+                rerank_heap::<O, _, _>(vector, prefetcher, fetch)
+                    .map(move |(distance, payload)| (opfamily.output(distance), payload)),
+            )
+        }
```

Comment:
```suggestion
        (RerankMethod::Heap, _) => {
            let prefetcher = PlainPrefetcher::<_, BinaryHeap<_>>::new(index, results);
            Box::new(
                rerank_heap::<O, _, _>(vector, prefetcher, fetch)
                    .map(move |(distance, payload)| (opfamily.output(distance), payload)),
            )
        }
```

It shouldn't be done.